packages/javascript/bh-shared-ui/src/components/AssetGroupFilters/AssetGroupFilters.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { AssetGroupMemberParams } from 'js-client-library/dist/types';
+import { FC, useState } from 'react';
+import { AzureNodeKind, ActiveDirectoryNodeKind, NodeIcon } from '../..';
+import {
+    Box,
+    Button,
+    Checkbox,
+    Collapse,
+    FormControl,
+    FormControlLabel,
+    Grid,
+    InputLabel,
+    MenuItem,
+    Paper,
+    Select,
+} from '@mui/material';
+import createStyles from '@mui/styles/createStyles';
+import makeStyles from '@mui/styles/makeStyles';
+import { Theme } from '@mui/material/styles';
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        formControl: {
+            display: 'block',
+        },
+        active: {
+            '& button': {
+                fontWeight: 'bolder',
+
+                '& span': {
+                    visibility: 'visible',
+                },
+            },
+        },
+        activeFilters: {
+            width: '6px',
+            height: '6px',
+            borderRadius: '100%',
+            backgroundColor: theme.palette.primary.main,
+            alignSelf: 'baseline',
+            visibility: 'hidden',
+        },
+    })
+);
+
+interface Props {
+    filterParams: AssetGroupMemberParams;
+    handleFilterChange: (
+        key: keyof Pick<AssetGroupMemberParams, 'primary_kind' | 'custom_member'>,
+        value: string
+    ) => void;
+    availableNodeKinds: Array<ActiveDirectoryNodeKind | AzureNodeKind>;
+}
+
+const AssetGroupFilters: FC<Props> = (props) => {
```

Comment:
Should we destructure these props?  Could make it easier for anyone else using this component to see what props it takes per intellisense.

Reply:
The props do get deconstructed just below this but it was a force of habit to not deconstruct in line with the function definition... however you are 100% right and most other component files deconstruct in line so Ill update to match! Thank you!